### PR TITLE
add test_ovs_restart_with_iperf_traffic

### PR DESCRIPTION
**Pull request by AlexandraAllakhverdieva**
1) Add class TestOVSRestartWithIperfTraffiс + test
2) Fix an issue with flavor in create_server() method

### CODE REVIEW

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 1]
```diff
@@ -480,3 +482,198 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+        return res_srv
+
+    def launch_iperf_client(self, client, server, keypair, vm_login,
+                            vm_pwd, background=False):
+        """Launch iperf client"""
+        client_cmd = 'iperf --port 5002 -u --client {0} --len 64' \
+                     ' --bandwidth 1M --time 60 -i 10' \
+            .format(self.os_conn.get_nova_instance_ips(server)['fixed'])
+        if background:
+            client_cmd = ' '.join(
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: Shorter way
`client_cmd += ' < /dev/null > ~/iperf_client.log 2>&1 &'`

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 2]
```diff
@@ -480,3 +482,198 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+            192.168.1.0/24, net02: net02__subnet, 192.168.2.0/24 and
+            attach them to router01.
+            4. Create keypair
+            5. Launch vm1 in net01 network and vm2 in net02 network
+            on different computes
+            6. Go to vm1 console and send pings to vm2
+        """
+
+        self.setup_rules_for_default_sec_group()
+        vm_image = self.create_image('/home/aallakhverdieva/images/',
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: What happens, if i haven't a folder '/home/aallakhverdieva/images/' or no file  'ubuntu-iperf.qcow2' in it?

Comment (AlexandraAllakhverdieva) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: Currently it is "IOError: [Errno 2] No such file or directory" and error as test result. But it's temporary solution (until we have no repository with all required images for tests).

Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: I see. In such case will be better to make it as a setting value (in mos-tests/settings.py) like other values. And add check, which skips this test, if image file not exists.
Also you may user `os.path.expanduser(~/images)` to get path relative to home folder.

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 3]
```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+        # make a list of all ovs agent ids
+        self.ovs_agent_ids = [agt['id'] for agt in
+                              self.os_conn.neutron.list_agents(
+                                  binary='neutron-openvswitch-agent')[
+                                  'agents']]
+
+        # make a list of ovs agents that resides only on controllers
+        controllers = [node.data['fqdn']
+                       for node in self.env.get_nodes_by_role('controller')]
+        ovs_agts = self.os_conn.neutron.list_agents(
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py:712]: Let's move `self.ovs_agent_ids =` below and rewrite it like:
`self.ovs_agent_ids = [agt['id'] for agt in ovs_agts]`

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 4]
```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+        lost_datagrams_rate_pattern = re.compile('\d+/\d+ \((\d+)%\)')
+        server_report_flag = False
+        for line in output:
+            if server_report_flag:
+                result = lost_datagrams_rate_pattern.search(line)
+                if result:
+                    return result.group(1)
+            elif line.endswith("Server Report:\n"):
+                server_report_flag = True
+        return False
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: Let's return None here

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 5]
```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+        :param output: list of lines (output of iperf client)
+        :return: percentage of lost datagrams
+        """
+        lost_datagrams_rate_pattern = re.compile('\d+/\d+ \((\d+)%\)')
+        server_report_flag = False
+        for line in output:
+            if server_report_flag:
+                result = lost_datagrams_rate_pattern.search(line)
+                if result:
+                    return result.group(1)
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: `return int(result.group(1))`

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 6]
```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+                                       vm_login='ubuntu', vm_pwd='ubuntu')
+        err_msg = 'Failed to start the iperf client on vm result: {}'.format(
+            res)
+        assert not res['exit_code'], err_msg
+
+        # Check iperf traffic before restart
+        lost = self.get_lost_percentage(res['stdout'])
+        err_msg = "Packet losses more than 1%. Actual value is {0}%".format(
+            lost)
+        assert not int(lost), err_msg
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]: `assert lost == 0, err_msg`

File: mos_tests/neutron/python_tests/test_ovs_restart.py [hunk 7]
```diff
@@ -480,3 +482,197 @@ def test_restart_openvswitch_agent_under_bat(self, prepare_openstack):
...
+        self.os_conn.wait_agents_down(self.ovs_conroller_agents)
+        self.restart_ovs_agents_on_computes()
+        self.enable_ovs_agents_on_controllers()
+        self.os_conn.wait_agents_alive(self.ovs_agent_ids)
+
+        cmd = 'cat ~/iperf_client.log'
+        result = self.run_on_vm(self.server1, self.instance_keypair, cmd,
+                                vm_login='ubuntu', vm_password='ubuntu')
+
+        while self.get_lost_percentage(result['stdout']) is False:
```
Comment (gdyuldin) [inline mos_tests/neutron/python_tests/test_ovs_restart.py]:
```
while True:
    result = ...
    if self.get_lost_percentage(result['stdout']) is not None:
        break
    time.sleep(5)
```

**Pull request merged**